### PR TITLE
Springdoc OpenAPI , HATEOAS, Flyway, Content Negotiation, Entity relations

### PR DESCRIPTION
**Pull request by nbugaenco**
This pull request introduces several enhancements and features to improve the functionality and usability of the application. Here's a summary of the changes made:

1. Added Entity Relations:
   - Implemented entity relationships to establish associations between entities, improving data consistency and integrity.

2. Added Flyway DB Migrations:
   - Implemented Flyway database migrations to manage and version the database schema.
   - This ensures that the database schema is up-to-date and can be easily migrated across different environments.

3. Optimized Models and UserService:
   - Performed optimization on models and UserService to enhance performance and code readability.
   - Reviewed and refactored the codebase for better efficiency and maintainability.

4. Added HATEOAS:
   - Implemented HATEOAS (Hypermedia as the Engine of Application State) to enable a self-describing API.
   - Added hyperlinks to the API responses, allowing clients to navigate and discover available actions and resources.

5. Removed PUT Mapping:
   - Removed the PUT mapping for updating users.
   - Instead, utilize PATCH mapping for partial updates, following best practices for RESTful APIs.

6. Added Springdoc OpenAPI:
   - Integrated Springdoc OpenAPI to generate comprehensive API documentation.
   - The API documentation is automatically generated from the codebase and provides detailed information about endpoints, request/response payloads, and error handling.

7. Added Content Negotiation (JSON/XML):
   - Implemented content negotiation to support both JSON and XML representations in API responses.
   - Clients can specify their preferred content type (JSON or XML) through the 'Accept' header, and the API will respond accordingly.

Please review and provide feedback on the changes made in this branch. The enhancements aim to improve the application's functionality, database management, code optimization, API documentation, and support for content negotiation.

### CODE REVIEW

File: src/main/java/com/eventsphere/user/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+     *
+     * @return ResponseEntity with the list of users and HTTP status OK.
+     */
+    @GetMapping
+    public ResponseEntity<CollectionModel<User>> getAllUsers() {
+        List<User> users = userService.getAll();
+
+        for (User user : users) {
+            user.add(
+                    linkTo(methodOn(UserController.class).getUser(user.getId())).withRel(GET_USER_REL),
```
Comment (vicol13) [inline src/main/java/com/eventsphere/user/controller/UserController.java:47]: what is happening here ?

File: src/main/java/com/eventsphere/user/exception/ErrorDetails.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+@Getter
+public class ErrorDetails {
+
+    private final LocalDateTime timestamp;
+    private final String message;
+    private final String details;
+
+    /**
+     * Constructs a new instance of {@code ErrorDetails}.
```
Comment (vicol13) [inline src/main/java/com/eventsphere/user/exception/ErrorDetails.java:18]: chat got generated documentation of constructor? use @Allargs. ...

**Comment by vicol13** [comment]
1. For future please keep merge request granular  (in aspect of added feature)  one merge per feature: 
    (Added Flyway DB Migrations: one request, Added Springdoc OpenAPI: second ) 

2. I can see duplicate code from previous merge request maybe you wanted to merge in the branch crud-feature 

3. You added spring-open-api dependency can you use annotations from library to document the endpoints

**Review by vicol13 — CHANGES_REQUESTED** [review]

File: src/main/java/com/eventsphere/user/controller/UserController.java (hunk 1 above)
Comment (nbugaenco) [reply src/main/java/com/eventsphere/user/controller/UserController.java:47]: Adding HATEOAS links to every user, so it will be easy to get the exact one.

**Pull request closed without merging**